### PR TITLE
PLATFORM-596: More general HSM container

### PR DESCRIPTION
**Pull request by mkrielza**
Allow container with CloudHSM client to start up without a valid HSM_IP.

### CODE REVIEW

File: docker/hsm.Dockerfile [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
 
+# Install SoftHSM2
+RUN yum install -y softhsm
+
+# Initialize some HSM slots as initial setup to be used in tests
+RUN softhsm2-util --init-token --slot 0 --label WALLET-000 --pin us3rs3cur3 --so-pin sup3rs3cur3
+RUN softhsm2-util --init-token --slot 1 --label WALLET-001 --pin us3rs3cur3 --so-pin sup3rs3cur3
+RUN softhsm2-util --init-token --slot 2 --label WALLET-002 --pin us3rs3cur3 --so-pin sup3rs3cur3
+RUN softhsm2-util --init-token --slot 3 --label WALLET-003 --pin us3rs3cur3 --so-pin sup3rs3cur3
+RUN softhsm2-util --init-token --slot 4 --label cavium --pin us3r:s3cur3 --so-pin sup3r:s3cur3
```
Comment (fllaca) [inline docker/hsm.Dockerfile]: this initialization is specific fixture for some testing stuff, it shouldn't be part of the base ethsigner image. We can create anothe Dockerfile somewhere else with something like this:

```Dockerfile
# adharaprojects/ethsinger:0.7.0-ADHARA-testfixtures
FROM adharaprojects/ethsinger:0.7.0-ADHARA

RUN softhsm2-util --init-token --slot 0 --label WALLET-000 --pin us3rs3cur3 --so-pin sup3rs3cur3
RUN softhsm2-util --init-token --slot 1 --label WALLET-001 --pin us3rs3cur3 --so-pin sup3rs3cur3
RUN softhsm2-util --init-token --slot 2 --label WALLET-002 --pin us3rs3cur3 --so-pin sup3rs3cur3
RUN softhsm2-util --init-token --slot 3 --label WALLET-003 --pin us3rs3cur3 --so-pin sup3rs3cur3
RUN softhsm2-util --init-token --slot 4 --label cavium --pin us3r:s3cur3 --so-pin sup3r:s3cur3
```

File: docker/hsm.Dockerfile [hunk 2]
```diff
@@ -1,6 +1,16 @@
 # Use the amazon linux image
 FROM amazonlinux:2
 
+# Install SoftHSM2
+RUN yum install -y softhsm
```
Comment (fllaca) [inline docker/hsm.Dockerfile]: Would this start any background "softhsm" service? I'm not familiar with this package, just asking. If softhsm2 will run a softhsm process I think we should maybe consider running it in separated containers, and make ethsigner connect to it over the network (if this is possible and makes sense for how softhsm works)

**Review by fllaca — COMMENTED** [review]
I have some doubts about bundling softhsm inside ethsigner image. It's fine if it's a client software to connect to a remote softhsm, but if it will start a backend softhsm process I think it shouldn't be part of eth-signer.

File: docker/cloudhsm-entrypoint.sh [hunk 3]
```diff
@@ -5,29 +5,29 @@
...
 if [[ -z "$HSM_IP" ]]; then
   #statements
-  echo -e "\n* HSM_IP env variable must be set \n"
-  exit 1
-fi
-
-# Configure cloudhms-client
-# COPY customerCA.crt /opt/cloudhsm/etc/
-/opt/cloudhsm/bin/configure -a $HSM_IP
+  echo -e "\n* HSM_IP env variable must be set to allow CloudHSM connectivity \n"
```
Comment (fllaca) [inline docker/cloudhsm-entrypoint.sh]:
```suggestion
  echo -e "\n* HSM_IP env variable not set: skipping CloudHSM connection \n"
```
more descriptive maybe of the actual flow happening?

**Review by fllaca — COMMENTED** [review]
Lgtm! Would you mind squashing commits into one? some intermediate changes aren't very meaningful to be kept in the git history

**Comment by mkrielza** [comment]
All SoftHSM parts was moved to an extended container in Dockerfiles. Only change left here is to not throw an error if you don't provide the HSM_IP when you start up the container. This is useful for the `all-in-one` extended ethsigner container that has SoftHSM capability. Thanks for the guidance Fer, it feels more right to not put it in the production container.

**Pull request closed without merging**